test/micro/org/openjdk/bench/vm/compiler/CountedLoopCastIV.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright (c) 2025, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package org.openjdk.bench.vm.compiler;
+
+import org.openjdk.jmh.annotations.*;
+
+import java.util.Objects;
+import java.util.Random;
+import java.util.concurrent.TimeUnit;
+
+@Warmup(iterations = 3, time = 1)
+@Measurement(iterations = 5, time = 1)
+@BenchmarkMode(Mode.Throughput)
+@OutputTimeUnit(TimeUnit.SECONDS)
+@State(Scope.Thread)
+@Fork(value=3)
+public class CountedLoopCastIV {
+    private static final int LEN = 2048;
+
+    private int start;
+    private int limit;
+    private int[] arr;
+
+    @Setup
+    public void init() {
+        arr = new int[LEN];
+        for (int i = 0; i < LEN; i++) {
+            arr[i] = i;
+        }
+
+        Random r = new Random();
+        start = r.nextInt(LEN >> 2);
+        limit = r.nextInt(LEN >> 1, LEN - 3);
```

Comment:
Does this not mean that we use a different seed every time, and therefore the loop has different lengths, and so the results can be influenced accordingly?

Reply:
Yes, I just want to make sure the loop length is different with each time running, so that it will not be influenced by some profiling related optimizations.

Reply:
I see. Did you have any direct issues with profiling here?

I'm worried that the trip count is really very variant here. We could have:
start = 0 and limit = LEN - 3      -> count ~ LEN
start = LEN/4 and limit = LEN/2  -> count = LEN/4

That is a 4x variance, am I right? Can you run the benchmark a few times, and see what the error term looks like?